machine/arch/src/main/java/org/qbicc/machine/arch/Cpu.java

```diff
@@ -28,10 +28,19 @@
     public static final Cpu PPC64 = new Cpu(8, "ppc64");
 
     private final int wordSize;
+    // TODO add an option to get this information from the user
+    private final int numProcessors;
+
+    Cpu(final int wordSize, final int numProcessors, final String name, final String... aliases) {
+        super(name, aliases);
+        this.wordSize = wordSize;
+        this.numProcessors = numProcessors;
```

Comment:
This would not be the correct place for this; the `Cpu` here gives information about the target CPU architecture, not the specific chip or run time configuration.  In practice, the available processors of the system will be governed by the operating system and can vary based on not only the physical device installed in the system but also the configuration of things like CGroups, etc.

The implementation of the `availableProcessors` method in qbicc will lazily compute the constant in the same manner that OpenJDK does.

Reply:
okay. reverting to my original workaround.